crates/nu-command/src/system/run_external.rs

```diff
@@ -41,8 +44,127 @@ impl Command for External {
         call: &Call,
         input: PipelineData,
     ) -> Result<PipelineData, ShellError> {
-        let command = create_external_command(engine_state, stack, call)?;
-        command.run_with_input(engine_state, stack, input, false)
+        let cwd = engine_state.cwd(Some(stack))?;
+
+        // Find the absolute path to the executable. On Windows, set the
+        // executable to "cmd.exe" if it's is a CMD internal command. If the
+        // command is not found, display a helpful error message.
+        let name: Spanned<String> = call.req(engine_state, stack, 0)?;
+        let executable = if cfg!(windows) && is_cmd_internal_command(&name.item) {
+            PathBuf::from("cmd.exe")
+        } else {
+            let paths = nu_engine::env::path_str(engine_state, stack, call.head)?;
+            let Some(executable) = which(&name.item, &paths, &cwd) else {
+                return Err(command_not_found(
+                    &name.item,
+                    call.head,
+                    engine_state,
+                    stack,
+                ));
+            };
+            executable
+        };
+
+        // Create the command.
+        let mut command = std::process::Command::new(executable);
+
+        // Configure PWD.
+        command.current_dir(cwd);
+
+        // Configure environment variables.
+        let envs = env_to_strings(engine_state, stack)?;
+        command.env_clear();
+        command.envs(envs);
+
+        // Configure args.
+        let args = eval_arguments_from_call(engine_state, stack, call)?;
+        if cfg!(windows) && is_cmd_internal_command(&name.item) {
+            // The /D flag disables execution of AutoRun commands from registry.
+            // The /C flag followed by a command name instructs CMD to execute
+            // that command and quit.
+            command.args(["/D", "/C", &name.item]);
+            // Check for special characters in `args` and reject them.
+            for arg in &args {
+                if has_cmd_special_character(&arg.item) {
+                    return Err(ShellError::ExternalCommand {
+                        label: "Special characters are not allowed in CMD builtins".into(),
+                        help: r#"These characters are special in CMD: / \ < > " | & ^"#.into(),
```

Comment:
This list makes me worry because I assume windows paths will be used for some internal commands like c:\program files and also quotes like "c:\program files". I have a slight concern about `/` being used as a parameter flag. The rest aren't quite as concerning. I thought we already escaped pipes with `^|`.

Reply:
Makes sense. I'll work on it.